### ISSUE TITLE
Helm installation ignores namespace

### ISSUE THREAD

**Issue by francoran** [senthilrch/kube-fledged#95]
Hey,

Trying to install via Helm and get the following error - 
```
helm upgrade -i kube-fledged-ml charts/kube-fledged \
            -f charts/kube-fledged/values.yaml \
            -n ml-services --atomic

Output - 

Release "kube-fledged-ml" does not exist. Installing it now.
Error: release kube-fledged-ml failed, and has been uninstalled due to atomic being set: namespaces "kube-fledged" not found
```
Even though I asked to install inside ml-services it's still looking for kube-fledged namespace. 

I also tried to override the namespace using --set kubefledgedNameSpace="ml-services"
This one did work but caused the actual ImageCache object to return the following error on creation - 

> Error from server (InternalError): error when creating "tmp-kube-fledged.yaml": Internal error occurred: failed calling webhook "validate-image-cache.kubefledged.io": Post "https://kube-fledged-ml-webhook-server.kube-fledged.svc:3443/validate-image-cache?timeout=1s": service "kube-fledged-ml-webhook-server" not found

Any ideas how to resolve?

**Comment by senthilrch** [comment]
I suspect when you installed the first time, the
validatingwebhookconfiguration would have been created with "kube-fledged"
as namespace. Since it gets created as a pre-install hook, helm delete will
not remove it. It has to be deleted using kubectl before doing another
install.

I suggest you list the validatingwebhookconfigurations in the cluster and
delete the old one.

**Comment by senthilrch** [comment]
I performed a helm install leaving behind the validatingwebhookconfiguration from a previous install. I get the same error you reported. I then deleted the old validatingwebhookconfiguration. After this creation of image cache is successful. Please see log below:-

```
eechens@EMB-Q6BUMD6N kube-fledged % helm upgrade -i kube-fledged-ml deploy/kubefledged-operator/helm-charts/kubefledged -n ml-services --atomic --set kubefledgedNameSpace="ml-services"
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /Users/eechens/.kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /Users/eechens/.kube/config
Release "kube-fledged-ml" does not exist. Installing it now.
NAME: kube-fledged-ml
LAST DEPLOYED: Mon Aug  9 22:33:45 2021
NAMESPACE: ml-services
STATUS: deployed
REVISION: 1
NOTES:
Kube-fledged installed successfully.
Refer to https://github.com/senthilrch/kube-fledged#how-to-use
Happy Image caching!!
eechens@EMB-Q6BUMD6N kube-fledged % kubectl get pods -n ml-services
NAME                                             READY   STATUS    RESTARTS   AGE
kube-fledged-ml-controller-548cc4b6fc-4hpzp      1/1     Running   0          62s
kube-fledged-ml-webhook-server-fc9d76d5c-66qmm   1/1     Running   0          62s
eechens@EMB-Q6BUMD6N kube-fledged % kubectl apply -f deploy/kubefledged-imagecache.yaml -n ml-services
Error from server (InternalError): error when creating "deploy/kubefledged-imagecache.yaml": Internal error occurred: failed calling webhook "validate-image-cache.kubefledged.io": Post https://kube-fledged-webhook-server.kube-fledged.svc:3443/validate-image-cache?timeout=1s: service "kube-fledged-webhook-server" not found
eechens@EMB-Q6BUMD6N kube-fledged % kubectl get validatingwebhookconfiguration
NAME                   WEBHOOKS   AGE
cert-manager-webhook   1          131d
kube-fledged           1          11d
kube-fledged-ml        1          3m12s
rancher.cattle.io      2          131d
eechens@EMB-Q6BUMD6N kube-fledged % kubectl delete validatingwebhookconfiguration kube-fledged
validatingwebhookconfiguration.admissionregistration.k8s.io "kube-fledged" deleted
eechens@EMB-Q6BUMD6N kube-fledged % kubectl apply -f deploy/kubefledged-imagecache.yaml -n ml-services
imagecache.kubefledged.io/imagecache1 created
eechens@EMB-Q6BUMD6N kube-fledged %
```

**Comment by senthilrch** [comment]
@francoran: Thanks for reporting this issue. The templates in the helm chart exclusively defaults to the namespace set in kubefledgedNameSpace. It should default to namespace supplied in the helm install command. Chart needs to be modified accordingly. Hoping to fix it in v0.8.2

**Comment by francoran** [comment]
Much appreciated, workaround did work. closing this issue

Thanks

**Comment by senthilrch** [comment]
The templates in the helm chart exclusively defaults to the namespace set in kubefledgedNameSpace. It should default to namespace supplied in the helm install command. Chart needs to be modified accordingly. Hoping to fix it in v0.8.2
